core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -57,6 +57,7 @@ private[spark] class TaskDescription(
     val addedJars: Map[String, Long],
     val addedArchives: Map[String, Long],
     val properties: Properties,
+    val cpus: Int,
```

Comment:
Add a precondition for `cpus` > 0 ?

Reply:
no precondition, just be consistent with other lines

Reply:
Please add a check to make sure we dont have invalid value for `cpus` coming in - `assert(cpus > 0)`